thread_pthread_mn.c

```diff
@@ -750,14 +861,51 @@ timer_thread_polling(rb_vm_t *vm)
             // simply retry
             break;
           default:
-            perror("epoll_wait");
-            rb_bug("epoll_wait errno:%d", errno);
+            perror("kq");
+            rb_bug("kq errno:%d", errno);
         }
         break;
 
       default:
         RUBY_DEBUG_LOG("%d event(s)", r);
 
+#if HAVE_SYS_EVENT_H
```

Comment:
This block of code is very similar to the epoll code, with small `kevent` specific changes. Maybe they should be consolidated more?